src/model/entities/rundown.ts

```diff
@@ -118,9 +118,15 @@ export class Rundown extends BasicRundown {
   }
 
   private findFirstSegment(): Segment {
-    return this.segments.reduce((previousSegment: Segment, currentSegment: Segment) => {
-      return previousSegment.rank < currentSegment.rank ? previousSegment : currentSegment
-    })
+    return this.segments
+      .filter(segment => this.isSegmentValidForRundownExecution(segment))
+      .reduce((previousSegment: Segment, currentSegment: Segment) => {
+        return previousSegment.rank < currentSegment.rank ? previousSegment : currentSegment
+      })
```

Comment:
There are other places in the the rundown entity (e.g. `findNextValidSegment`) where we assume that the segments list is ordered correctly – why can't we do that here?

Reply:
I think that is a leftover from the initial implementation and is no longer needed since we keep the Segments sorted.
Updated to 
```
return this.segments.filter(segment => this.isSegmentValidForRundownExecution(segment))[0]
```
which is not great either incase there are no Segments. 
Suggestions are appreciated.